core-rust/state-manager/src/mempool/priority_mempool.rs

```diff
@@ -430,79 +396,53 @@ impl PriorityMempool {
             .sub(transaction_size as i64);
         self.metrics.current_transactions.sub(1);
 
-        self.data.remove(payload_hash);
+        self.data.remove(notarized_transaction_hash);
 
         // Update intent_lookup
         let payload_lookup = self
             .intent_lookup
             .get_mut(intent_hash)
             .expect("Mempool intent hash lookup out of sync on remove");
 
-        if !payload_lookup.remove(payload_hash) {
+        if !payload_lookup.remove(notarized_transaction_hash) {
             panic!("Mempool intent hash lookup out of sync on remove");
         }
         if payload_lookup.is_empty() {
             self.intent_lookup.remove(intent_hash);
         }
 
-        // Remove from proposal_priority_index
-        if !self
-            .proposal_priority_index
-            .remove(&MempoolDataProposalPriorityOrdering(data.clone()))
-        {
-            panic!("Mempool priority index out of sync on remove");
-        }
-
-        // Remove from end_epoch_exclusive_index
-        if !self
-            .end_epoch_exclusive_index
-            .remove(&MempoolDataEndEpochExclusiveOrdering(data.clone()))
-        {
-            panic!("Mempool end epoch index out of sync on remove");
-        }
-
-        // Remove from executed_state_version_index
-        let state_version = match self.data_to_executed_state_version.remove(&data) {
-            None => panic!("Mempool state version map out of sync on remove"),
-            Some(state_version) => state_version,
-        };
-
-        if !self
-            .executed_state_version_index
-            .remove(&MempoolDataStateVersion(state_version, data))
-        {
-            panic!("Mempool executed state version index out of sync on remove");
-        }
+        self.proposal_priority_index
+            .remove(ProposalPriority::new(&data), *notarized_transaction_hash);
```

Comment:
We should add an `expect(..)` to all of these to panic if for some reason the indices get out of line.

EDIT: Ahh I see this is inline on the `remove` method. All good! Perhaps `remove_expected_entry` might be clearer, but `remove` is fine too.

Reply:
![image](https://github.com/radixdlt/babylon-node/assets/119413677/511516e9-a61c-4054-83bc-d6c1cafc92a1)
... and this has to be accompanied by `insert_unique()` 👍

Reply:
Beautiful name suggestions :D